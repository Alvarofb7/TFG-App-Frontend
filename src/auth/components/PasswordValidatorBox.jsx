export const PasswordValidatorBox = ({
	lowerValid,
	upperValid,
	lengthValid,
	numberValid,
	specialCharValid,
	password,
}) => {
	return (
		<main
			className="password-validation-box"
			aria-label="password-validation-box"
			style={{
				display:
					(lengthValid &&
						lowerValid &&
						upperValid &&
						numberValid &&
						specialCharValid) ||
					password === ""
						? "none"
						: "",
			}}
		>
			<div className={lengthValid ? "validated" : "not-validated"}>
				{lengthValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Debe de contener entre 6 y 20 caracteres
			</div>
			<div className={lowerValid ? "validated" : "not-validated"}>
				{lowerValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos una letra minúscula
			</div>
			<div className={upperValid ? "validated" : "not-validated"}>
				{upperValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos una letra mayúscula
			</div>
			<div className={numberValid ? "validated" : "not-validated"}>
				{numberValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos un número
			</div>
			<div className={specialCharValid ? "validated" : "not-validated"}>
				{specialCharValid ? (
					<span className="list-icon green">
						<i className="fa-solid fa-check"></i>
					</span>
				) : (
					<span className="list-icon">
						<i className="fa-solid fa-xmark"></i>
					</span>
				)}
				Al menos un caracter especial
			</div>
		</main>
	);
};
