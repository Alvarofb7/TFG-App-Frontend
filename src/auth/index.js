export * from "./components/PasswordValidatorBox";
export * from "./pages/LoginPage";
export * from "./layout/HomePage";
