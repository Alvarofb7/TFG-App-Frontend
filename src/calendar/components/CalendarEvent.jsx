export const CalendarEvent = ({ event }) => {
	const { title, notes } = event;

	return (
		<span>
			<strong>{title}</strong> {notes && <span>- {notes}</span>}{" "}
		</span>
	);
};
