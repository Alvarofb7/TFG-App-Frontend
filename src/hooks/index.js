
export * from "./useAuthStore";
export * from "./useCalendarStore";
export * from "./useForm";
export * from "./useNoteStore";
export * from "./useUiStore";