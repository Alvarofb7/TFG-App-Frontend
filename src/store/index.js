export * from "./auth/authSlice";
export * from "./calendar/calendarSlice";
export * from "./kanban/kanbanSlice";
export * from "./notes/notesSlice";

export * from "./store";

