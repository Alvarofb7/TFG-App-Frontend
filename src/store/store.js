import { configureStore } from "@reduxjs/toolkit";
import { authSlice, calendarSlice, kanbanSlice, notesSlice, uiSlice } from "./";


export const store = configureStore({
    reducer: {
        auth: authSlice.reducer,
        calendar: calendarSlice.reducer,
        notes: notesSlice.reducer,
        kanban: kanbanSlice.reducer,
        ui: uiSlice.reducer,
    },
    middleware: (getDefaultMiddleware) => getDefaultMiddleware({
        serializableCheck: false
    })
})