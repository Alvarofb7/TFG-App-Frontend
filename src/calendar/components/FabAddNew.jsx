import { addHours } from "date-fns";
import { useCalendarStore, useUiStore } from "../../hooks";

export const FabAddNew = () => {

    const { openDateCalendarModal } = useUiStore();
    const { setActiveEvent } = useCalendarStore();

    const handleClick = () => {
        setActiveEvent({
            title: "",
            notes: "",
            allDay: false,
            start: new Date(),
            end: addHours(new Date(), 2),
            user: {}
        })
        openDateCalendarModal();
    }

    return (
        <button
            className="btn btn-primary fab"
            onClick={ handleClick }
        >
            <i className="fas fa-plus"></i>
        </button>
    )
}
