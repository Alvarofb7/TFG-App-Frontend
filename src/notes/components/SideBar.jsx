import { useEffect } from "react";
import { useSelector } from "react-redux";

import { useNoteStore } from "../../hooks";
import { RenderNotes } from "./RenderNotes";

export const SideBar = ({ show }) => {
	const { notes, isLoadingNotes } = useSelector((state) => state.notes);
	const { startLoadingNote, setActiveNewNote } = useNoteStore();

	useEffect(() => {
		startLoadingNote();
	}, []);

	const handleClick = () => {
		setActiveNewNote();
	};

	return (
		<nav
			className="nav-list-notes"
			id="list"
			style={{ display: show }}
			aria-label="sideBar"
		>
			<div className="list-note-button">
				<button
					className="btn btn-create-note"
					type="button"
					onClick={handleClick}
					aria-label="btn-add-new-note"
				>
					Crear nota
				</button>
			</div>
			<ul className="list-notes list-group">
				{!isLoadingNotes ? (
					<RenderNotes notes={notes} />
				) : (
					<li className="list-group-item">Cargando notas...</li>
				)}
			</ul>
		</nav>
	);
};
