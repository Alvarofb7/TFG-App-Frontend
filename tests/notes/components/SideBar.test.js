import { fireEvent, render, screen } from "@testing-library/react";
import { useNoteStore } from "../../../src/hooks";
import { SideBar } from "../../../src/notes/components/SideBar";
import { notesSlice } from "../../../src/store";
import {
	noteInitialState,
	noteWithId,
	noteWithNoteState,
} from "../../fixtures/notesStates";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";

jest.mock("../../../src/hooks/useNoteStore");

const getMockStore = (initialState) => {
	return configureStore({
		reducer: {
			notes: notesSlice.reducer,
		},
		preloadedState: {
			notes: { ...initialState },
		},
	});
};

describe("Pruebas del componente SideBar", () => {
	const mockStartLoadingNote = jest.fn();
	const mockSetActiveNewNote = jest.fn();

	beforeEach(() => {
		jest.clearAllMocks();
		jest.clearAllTimers();

		useNoteStore.mockReturnValue({
			setActiveNewNote: mockSetActiveNewNote,
			startLoadingNote: mockStartLoadingNote,
		});
	});

	test("No debe de mostrarse el componente", () => {
		const mockStore = getMockStore(noteInitialState);

		render(
			<Provider store={mockStore}>
				<SideBar show={"none"} />
			</Provider>
		);

		const sideBar = screen.getByLabelText("sideBar");

		expect(sideBar.style.display).toBe("none");
		expect(mockStartLoadingNote).toHaveBeenCalled();
	});

	test("Debe de mostrarse el componente correctamente", () => {
		const mockStore = getMockStore(noteInitialState);

		const { container } = render(
			<Provider store={mockStore}>
				<SideBar show={""} />
			</Provider>
		);

		expect(container).toMatchSnapshot();
		expect(mockStartLoadingNote).toHaveBeenCalled();
	});

	test("Deben de mostrarse los eventos en el componente correctamente", () => {
		const mockStore = getMockStore(noteWithNoteState);

		render(
			<Provider store={mockStore}>
				<SideBar show={""} />
			</Provider>
		);

		const li = screen.getByLabelText(`note-${noteWithId.id}`);
		expect(li).toBeInTheDocument();
		expect(mockStartLoadingNote).toHaveBeenCalled();
	});

	test("Deben de llamar a setActiveNewNote cuando se haga click en el botÃ³n", () => {
		const mockStore = getMockStore(noteWithNoteState);

		render(
			<Provider store={mockStore}>
				<SideBar show={""} />
			</Provider>
		);

		const btn = screen.getByLabelText("btn-add-new-note");
		fireEvent.click(btn);

		expect(mockSetActiveNewNote).toHaveBeenCalled();
	});
});
