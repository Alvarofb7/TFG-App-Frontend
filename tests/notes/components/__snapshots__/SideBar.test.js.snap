// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente SideBar Debe de mostrarse el componente correctamente 1`] = `
<div>
  <nav
    aria-label="sideBar"
    class="nav-list-notes"
    id="list"
  >
    <div
      class="list-note-button"
    >
      <button
        aria-label="btn-add-new-note"
        class="btn btn-create-note"
        type="button"
      >
        Crear nota
      </button>
    </div>
    <ul
      class="list-notes list-group"
    >
      <li
        class="list-group-item"
      >
        Cargando notas...
      </li>
    </ul>
  </nav>
</div>
`;
